ontology/rml-sourcetarget.owl

```diff
@@ -0,0 +1,267 @@
+
+
+###  http://www.linkedmodel.org/schema/vaem#acronym
+vaem:acronym rdf:type owl:AnnotationProperty .
+
+
+###  http://xmlns.com/foaf/0.1/name
+foaf:name rdf:type owl:AnnotationProperty .
+
+
+#################################################################
+#    Object Properties
+#################################################################
+
+###  http://w3id.org/rml/compression
+:compression rdf:type owl:ObjectProperty ;
+             rdfs:domain [ rdf:type owl:Class ;
+                           owl:unionOf ( :Source
+                                         :Target
+                                       )
+                         ] ;
+             rdfs:range comp:Compression ;
+             rdfs:comment <http://w3id.org/rml/source-target/> ;
+             rdfs:label "compression" .
+
+
+###  http://w3id.org/rml/encoding
+:encoding rdf:type owl:ObjectProperty ;
+          rdfs:domain [ rdf:type owl:Class ;
+                        owl:unionOf ( :Source
+                                      :Target
+                                    )
+                      ] ;
+          rdfs:range enc:Encoding ;
+          rdfs:comment <http://w3id.org/rml/source-target/> ;
+          rdfs:label "encoding" .
+
+
+###  http://w3id.org/rml/iterator
+:iterator rdf:type owl:ObjectProperty ;
+          rdfs:domain :LogicalSource ;
+          rdfs:range rdf:Literal ;
+          rdfs:comment <http://w3id.org/rml/source-target/> ;
+          rdfs:label "iterator" .
+
+
+###  http://w3id.org/rml/logicalSource
+:logicalSource rdf:type owl:ObjectProperty ;
+               rdfs:domain :TriplesMap ;
+               rdfs:range :LogicalSource ;
+               rdfs:comment "Represents the logical source to be mapped. This can be a pointer to any dataset."@en ;
+               rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+               rdfs:label "logical source" .
+
+
+###  http://w3id.org/rml/logicalTarget
+:logicalTarget rdf:type owl:ObjectProperty ;
+               rdfs:domain :TermMap ;
+               rdfs:range :LogicalTarget ;
+               rdfs:comment "Represents the logical target to where triples are exported to. This can be a pointer to any dataset." ;
+               rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+               rdfs:label "logical target" .
+
+
+###  http://w3id.org/rml/null
+:null rdf:type owl:ObjectProperty ;
+      rdfs:domain :Source ;
+      rdfs:range rdf:Literal ;
+      rdfs:comment <http://w3id.org/rml/source-target/> ;
+      rdfs:label "null" .
+
+
+###  http://w3id.org/rml/query
+:query rdf:type owl:ObjectProperty ;
+       rdfs:domain :Source ;
+       rdfs:range rdf:Literal ;
+       rdfs:comment <http://w3id.org/rml/source-target/> ;
+       rdfs:label "query" .
+
+
+###  http://w3id.org/rml/referenceFormulation
+:referenceFormulation rdf:type owl:ObjectProperty ;
+                      rdfs:domain :LogicalSource ;
+                      rdfs:range :ReferenceFormulation ;
```

Comment:
rml:referenceFormulation takes an IRI which can be anything. We provide some reference formulations via the QL namespace, but anything is open here for extensibility.

Reply:
Similar to Encoding & Compression

Reply:
Ah maybe they are... There's this:

```
rml:referenceFormulation [ a ql:XPathReferenceFormulation;
       ql:namespace [ a ql:Namespace;
         ql:namespacePrefix "ex";
         ql:namespaceURL "http://example.org";
       ];
     ];
```

(misses in the ontology + diagram though)

Reply:
So we have this `rml:ReferenceFormulation` class right?
(needed for shapes)

This can be done then:

```
rml:referenceFormulation [ a ql:XPathReferenceFormulation;
       ql:namespace [ a ql:Namespace;
         ql:namespacePrefix "ex";
         ql:namespaceURL "http://example.org";
       ];
     ];
```

But what about:

```
rml:referenceFormulation ql:XPath;
```

Does that still work then as well semantic/ontology-wise? 
(needed to adjust the spec)

@anaigmo @pmaria

Reply:
> The way I see it, ql:XPathReferenceFormulation seems to be a subclass of ql:ReferenceFormulation, and is the only one with the property ql:Namespace. It's no problem adding this, but isn't there any other case of a reference formulation needing ql:namespace?

I thinks that's correct. The other reference formulations are also subclasses, but without any properties (I think!).
AFAIK, namespaces are only applicable to XML, JSONPath does not have that. For me, reference formulations are pretty open, to allow new reference formulations in the future. These future things may require even different properties.